contents/javascript/VueJs/Intro.md

```diff
@@ -0,0 +1,134 @@
+
+## Creating a Simple Project in VueJs:
+
+The VueJs version of "Hello World":
+
+```HTML
+<!--This is the main HTML file-->
+<body>
+  <div id="root">
+		<h1>{{ message }}</h1>
+	</div>
+	<script src="https://unpkg.com/vue@2.5.13/dist/vue.js"></script>
+	<script src="the_path_to_the_javacript_file.js"></script>
+</body>
+```
+
+```js
+// This is inside the javacript file
+new Vue ({
+	el: '#root',
+
+  data: {
+    message: "Hello World"
+  }
+});
+```
+
+Explanation:
+* Import VueJs CDN and the JavaScript file in the main HTML file.
+* Create a Vue instance in the JavaScript file; bind the instance to one of the component in our html file (the `root` element). (In this case, only this component is working in VueJs while the rest are unaffected. This is how we progressively plug in VueJs into our projects without a complete one at a go.)
+* Specify our data(message: "Hello World") in the Vue instance.
+* Pass the message to the HTML file using double curly brackets.
+Remark: we can only use the Vue data inside the component with id `root` as we bind it to the Vue element.
+Open the brower and we will see "Hello World" being displayed.
+
+### Common VueJs syntax
+
+* [Text](https://vuejs.org/v2/guide/syntax.html#Text)
+The most basic data binding is text interpolation using double curly brackets as shown in the above example. The mustache tag will be replaced with the value of the msg property on the corresponding data object. It will also be updated whenever the data object’s msg property changes. The double curly brackets as the identifier inside it will be replaced with the value of the `message` property on the corresponding data object. It will also be updated whenever the data object’s `message` property changes.
+
+* [Attributes](https://vuejs.org/v2/guide/syntax.html#Attributes)
+Double curly brackets does not work in a HTML attributes. Instead, use a `v-bind` directive.
+```HTML
+<div v-bind:id="dynamicId"></div>
+```
+In the case of boolean attributes, where their mere existence implies true, v-bind works a little differently. In this example:
+```HTML
+<button v-bind:disabled="isButtonDisabled">Button</button>
+```
+If `isButtonDisabled` has the value of `null`, `undefined`, or `false`, the disabled attribute will not even be included in the rendered `<button>` element.
+
+* [Directives](https://vuejs.org/v2/guide/syntax.html#Directives)
+Directives are special attributes with the `v-` prefix. Directive attribute values are expected to be a single JavaScript expression (with the exception for `v-for`). A directive’s job is to reactively apply side effects to the DOM when the value of its expression changes.
+Example of how to use the directives are as follow:
+
+  - `v-example="value"` - this will pass a value into the directive, and the directive will response according to the value.
```

Comment:
What kinds of attributes can directives have? I don't think the `v-example="value"` syntax is very clear. Maybe you can explicitly say that this is the format in which directives are written.